src/KingSystem/System/Quest.cpp

```diff
@@ -0,0 +1,116 @@
+#include "KingSystem/System/Quest.h"
+
+namespace ksys::qst {
+
+// chunks missing
+Quest::~Quest() {
+    mSteps.freeBuffer();
+}
+
+// NON_MATCHING
+Quest::Quest(const sead::SafeString& name, sead::Heap* heap) : mName(name), mHeap(heap) {
+    _8 = 0;
+    _c = 0;
+    _10 = 0;
+    mNameHash = agl::utl::ParameterBase::calcHash(mName);
+    mAocVersionFlags = 0;
+}
+
+void Quest::initFlags(gdt::Manager* gdm) {
+    if (gdm == nullptr)
+        return;
+
+    mReady = gdm->getBoolHandle([&] {
+        sead::FixedSafeString<64> ready;
+        ready.format("%s_Ready", mName.cstr());
+        return ready;
+    }());
+
+    mCancelled = gdm->getBoolHandle([&] {
+        sead::FixedSafeString<64> cancelled;
+        cancelled.format("%s_Cancelled", mName.cstr());
+        return cancelled;
+    }());
+
+    switch (mQuestDependencyFlagType) {
+    default:
+        mDependencyFlag = gdt::InvalidHandle;
+        break;
+    case 0:
+        mDependencyFlag = gdm->getBoolHandle(mQuestDependencyFlag);
+        break;
+    case 1:
+        mDependencyFlag = gdm->getF32Handle(mQuestDependencyFlag);
+        break;
+    case 2:
+        mDependencyFlag = gdm->getS32Handle(mQuestDependencyFlag);
+        break;
+    }
+}
+
+void Quest::setField31() {
+    _31 = 1;
+}
+
+bool Quest::x_1() const {
+    bool result = false;
+
+    if ((gdt::Manager::instance() == nullptr) || (mCancelled == gdt::InvalidHandle))
```

Comment:
parentheses not required
```suggestion
    if (gdt::Manager::instance() == nullptr || mCancelled == gdt::InvalidHandle)
```